packages/utilities/src/styled.test.tsx

```diff
@@ -100,4 +101,33 @@ describe('styled', () => {
     const component = renderer.create(<Test cool />);
     expect(component.toJSON()).toMatchSnapshot();
   });
+
+  it('can produce different styles object if not cached', () => {
+    const component = renderer.create(<Test cool />);
+    const originStyles = _lastProps!.styles;
+
+    component.update(<Test cool />);
+    expect(_lastProps!.styles).not.toBe(originStyles);
+  });
+
+  it('can produce same styles object if cached', () => {
+    const component = renderer.create(<TestCached cool />);
+    expect(component.toJSON()).toMatchSnapshot();
+    let originStyles = _lastProps!.styles;
+
+    component.update(<TestCached cool />);
+    expect(component.toJSON()).toMatchSnapshot();
+    expect(_lastProps!.styles).toBe(originStyles);
+    originStyles = _lastProps!.styles;
+
+    const styles = { root: { color: 'red' } };
+    component.update(<TestCached cool styles={styles} />);
+    expect(component.toJSON()).toMatchSnapshot();
+    expect(_lastProps!.styles).not.toBe(originStyles);
+    originStyles = _lastProps!.styles;
+
+    component.update(<TestCached styles={styles} />);
+    expect(component.toJSON()).toMatchSnapshot();
+    expect(_lastProps!.styles).toBe(originStyles);
```

Comment:
I expect this line NOT to match the previous, when passing in a function which recalculates styles and changes based on the color.

Reply:
I have no idea why this should NOT match the previous, as `props.styles` is actually a __FUNCTION__ that will be invoked during rendering. 

If some props changes as you mentioned, the component will render, and `props.styles` get called during rendering. The styles function do not need to change, it will react to props changes and produce different output even if the function itself is not changed. So everything works as expected.

I'm not sure that is there any reason to expect `props.styles` to change?

Reply:
Ok, I think I was emphasizing the wrong thing here. If you mutate props, I expect the styles to be re-evaluated. Updated your tests to add some cases that are failing now. See my update and let me know if that makes more sense.